staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -221,6 +260,7 @@ func (r *withRetry) Before(ctx context.Context, request *Request) error {
 	// apiserver at least once before. This request should
 	// also be throttled with the client-internal rate limiter.
 	if err := request.tryThrottleWithInfo(ctx, r.retryAfter.Reason); err != nil {
+		r.trackPreviousError(ctx.Err())
```

Comment:
why not `return WrapPreviousError(ctx.Err())` and the callers don't have to wrap?

Reply:
> why not `return WrapPreviousError(ctx.Err())` and the callers don't have to wrap?

perhaps is just for congruency?